plugins/aws/sts_provisioner.go

```diff
@@ -16,10 +16,33 @@ import (
 const defaultProfileName = "default"
 
 type STSProvisioner struct {
-	profileName string
+	profileName     string
+	providerFactory STSProviderFactory
```

Comment:
🙌 for adding the tests that use this. 

A small suggestion to simplify the interface a bit:
```suggestion
    newProviderFactory func(cacheState sdk.CacheState, cache sdk.CacheOperations, fields map[sdk.FieldName]string) STSProviderFactory
```

I think that allows the removal of the `SetState()` method and thereby makes it impossible to misuse the interface (calling `NewAssumeRoleProvider()` before `SetState()` is illegal in the current form).

However, I don't see any big issues with the current form. So you're also free to leave it as-is.